### ISSUE TITLE
Style/DateTime marks usages of DateTime.iso8601

### ISSUE THREAD

**Issue by NobodysNightmare** [rubocop-hq/rubocop#4925]
## Steps to reproduce the problem

Run rubocop over the following snippet:

```ruby
DateTime.iso8601(iso_string)
```
## Actual behavior

Rubocop registers a `Style/DateTime` offense and proposes to use `Date` or `Time` instead.

## Expected behavior

In my opinion this case should not generate an offense, since neither `Date` nor `Time` offer the same format compatibility as `DateTime` (see "Behavioural comparison" below).

## Behavioural comparison

This irb snippet shows the difference between the three classes:

```ruby
irb(main):001:0> require 'time'
irb(main):002:0> date_string = '2017-01-01'
irb(main):003:0> time_string = '2017-01-01T12:34:56Z'

irb(main):004:0> Time.iso8601(date_string)
ArgumentError: invalid date: "2017-01-01"
irb(main):005:0> Time.iso8601(time_string)
=> 2017-01-01 12:34:56 UTC

irb(main):006:0> Date.iso8601(date_string)
=> #<Date: 2017-01-01 ((2457755j,0s,0n),+0s,2299161j)>
irb(main):007:0> Date.iso8601(time_string)
=> #<Date: 2017-01-01 ((2457755j,0s,0n),+0s,2299161j)>

irb(main):008:0> DateTime.iso8601(date_string)
=> #<DateTime: 2017-01-01T00:00:00+00:00 ((2457755j,0s,0n),+0s,2299161j)>
irb(main):009:0> DateTime.iso8601(time_string)
=> #<DateTime: 2017-01-01T12:34:56+00:00 ((2457755j,45296s,0n),+0s,2299161j)>
```

While `Date` will not care whether it gets a long or a short ISO string, it of course ignores the time portion.

The `Time` class refuses accepting a short ISO string.

Only `DateTime` will accept both.

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.51.0 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-linux)
```

**Comment by Drenmi** [comment]
We should not auto-correct this, as it would break compatibility.

I would be rather worried though, if the developer does not know what result they are expecting. Using `DateTime` and not getting any time part looks like it will lead to a silent (and potentially costly) bug. 🙂

**Comment by bbatsov** [comment]
I'm closing this ticket due to no recent activity. Feel free to re-open it if you ever come back to it.
